client/src/components/client-dashboard-component/FilterCard.jsx

```diff
@@ -0,0 +1,226 @@
+          label="From"
+          type="date"
+          value={values.dateUpdatedFrom || ""}
+          onChange={handleChange("dateUpdatedFrom")}
+          margin="normal"
+          InputLabelProps={{
+            shrink: true,
+          }}
+        />
+        <TextField
+          label="Until"
+          type="date"
+          value={values.dateUpdatedUntil || ""}
+          onChange={handleChange("dateUpdatedUntil")}
+          margin="normal"
+          InputLabelProps={{
+            shrink: true,
+          }}
+        />
+        <Typography variant="body1" sx={{ textAlign: "left" }}>
+          Date Registered
+        </Typography>
+        <TextField
+          label="From"
+          type="date"
+          value={values.dateRegisteredFrom || ""}
+          onChange={handleChange("dateRegisteredFrom")}
+          margin="normal"
+          InputLabelProps={{
+            shrink: true,
+          }}
+        />
+        <TextField
+          label="Until"
+          type="date"
+          value={values.dateRegisteredUntil || ""}
+          onChange={handleChange("dateRegisteredUntil")}
+          margin="normal"
+          InputLabelProps={{
+            shrink: true,
+          }}
+        />
+        <Typography variant="body1" sx={{ textAlign: "left" }}>
+          Owner
+        </Typography>
+        <Select
+          labelId="owner-label"
+          id="owner-select"
+          value={values.owner}
+          onChange={handleChange("owner")}
+          fullWidth
+        >
+          <MenuItem value="All">All</MenuItem>
+        </Select>
+        <Typography variant="body1" sx={{ textAlign: "left" }}>
+          Status
+        </Typography>
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={values.status.active}
+              onChange={handleStatusChange("active")}
+            />
+          }
+          label="Active"
+        />
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={values.status.rAndI}
+              onChange={handleStatusChange("rAndI")}
+            />
+          }
+          label="R&I"
+        />
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={values.status.closed}
+              onChange={handleStatusChange("closed")}
+            />
+          }
+          label="Closed"
+        />
+        {values.status.closed && (
+          <Box sx={{ pl: 3 }}>
+            <RadioGroup
+              aria-label="action-status"
+              name="actionStatus"
+              value={values.actionStatus}
+              onChange={handleChange("actionStatus")}
+            >
+              <FormControlLabel value="all" control={<Radio />} label="All" />
+              <FormControlLabel
+                value="actionNeeded"
+                control={<Radio />}
+                label="Action Needed"
+              />
+            </RadioGroup>
+          </Box>
+        )}
```

Comment:
Add a button for applying filters as well
<img width="165" alt="Screenshot 2024-01-26 at 12 31 18 PM" src="https://github.com/uoftblueprint/tces/assets/58835213/6b4af1e9-54cd-46d1-b492-f05c36f8a989">

Reply:
If you would like, you can copy the way Kevin has those buttons implemented on the job-leads-table. Otherwise these button will be cleaned up in a small ticket later on